src/outputs/parthenon_hdf5.cpp

```diff
@@ -415,17 +462,26 @@ void PHDF5Output::WriteOutputFileImpl(Mesh *pm, ParameterInput *pin, SimTime *tm
     ~MPI_InfoDeleter() { MPI_Info_free(&info); }
   } delete_info{FILE_INFO_TEMPLATE};
 
-  PARTHENON_HDF5_CHECK(H5Pset_sieve_buf_size(acc_file, 262144));
-  PARTHENON_HDF5_CHECK(H5Pset_alignment(acc_file, 524288, 262144));
-
-  PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "access_style", "write_once"));
-  PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "collective_buffering", "true"));
-  PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "cb_block_size", "1048576"));
-  PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "cb_buffer_size", "4194304"));
+  // Hint specifies the manner in which the file will be accessed until the file is closed
+  char* access_style = Env::get<char *>("MPI_access_style", (char*)"write_once", exists);
+  PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "access_style", access_style));
+
+  // Specifies whether the application may benefit from collective buffering
+  // Default :: collective_buffering is disabled
+  bool collective_buffering = Env::get<bool>("MPI_collective_buffering", false, exists);
+  if(collective_buffering) {
+    PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "collective_buffering", "true"));
+    // Specifies the block size to be used for collective buffering file acces
+     char *cb_block_size = Env::get<char *>("MPI_cb_block_size", (char*)"1048576", exists);
+    PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "cb_block_size",  cb_block_size));
+    // Specifies the total buffer space that can be used for collective buffering on each target node, 
+    // usually a multiple of cb_block_size
+    char *cb_buffer_size = Env::get<char *>("MPI_cb_buffer_size", (char*)"4194304", exists);
+    PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "cb_buffer_size", cb_buffer_size));
+  }
```

Comment:
Why did the logic on how to handle env vars change here compared to above (where `exits` would be use to distinguish between default and not set at all)?

Reply:
fixed to use exists